File: test/language/templates/cpp98/InstantiateVsDefaultTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_vs_default/InstantiateVsDefault.h"

namespace templates
{
namespace instantiate_vs_default
{

TEST(InstantiateVsDefaultTest, readWrite)
{
    InstantiateVsDefault instantiateVsDefault;
    pkg::Test_uint32 u32;
    u32.setValue(13);
    instantiateVsDefault.setTest32(u32);
    TStr str;
    str.setValue("test");
    instantiateVsDefault.setTestStr(str);

    zserio::BitStreamWriter writer;
    instantiateVsDefault.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateVsDefault readInstantiateVsDefault(reader);

    ASSERT_TRUE(instantiateVsDefault == readInstantiateVsDefault);
}

} // namespace instantiate_vs_default
} // namespace templates

File: test/language/templates/java/templates/InstantiateVsDefaultTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;

import java.io.IOException;

import org.junit.jupiter.api.Test;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_vs_default.InstantiateVsDefault;
import templates.instantiate_vs_default.TStr;
import templates.instantiate_vs_default.pkg.Test_uint32;

public class InstantiateVsDefaultTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateVsDefault instantiateVsDefault = new InstantiateVsDefault();
        instantiateVsDefault.setTest32(new Test_uint32(13));
        instantiateVsDefault.setTestStr(new TStr("test"));

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        instantiateVsDefault.write(writer);

        final BitStreamReader reader =
                new ByteArrayBitStreamReader(writer.toByteArray(), writer.getBitPosition());
        final InstantiateVsDefault readInstantiateVsDefault = new InstantiateVsDefault(reader);
        assertTrue(instantiateVsDefault.equals(readInstantiateVsDefault));
    }
}
